File: mifosng-android/src/main/java/com/mifos/objects/templates/savings/SavingProductsTemplate.java
package com.mifos.objects.templates.savings;

import com.mifos.objects.Currency;
import com.mifos.objects.InterestType;
import com.mifos.objects.PaymentTypeOption;

import java.util.List;

/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

/**
 * Created by rajan on 13/3/16.
 */
public class SavingProductsTemplate {

    private Currency currency;
    private InterestType interestCompoundingPeriodType;
    private InterestType interestPostingPeriodType;
    private InterestType interestCalculationType;
    private InterestType interestCalculationDaysInYearType;
    private InterestType accountingRule;
    private List<Currency> currencyOptions;
    private List<InterestType> interestCompoundingPeriodTypeOptions;
    private List<InterestType> interestPostingPeriodTypeOptions;
    private List<InterestType> interestCalculationTypeOptions;
    private List<InterestType> interestCalculationDaysInYearTypeOptions;
    private List<InterestType> lockinPeriodFrequencyTypeOptions;
    private List<InterestType> withdrawalFeeTypeOptions;
    private List<PaymentTypeOption> paymentTypeOptions;
    private List<InterestType> accountingRuleOptions;
    private AccountOptions liabilityAccountOptions;
    private List<AccountOptions> assetAccountOptions;
    private List<AccountOptions> expenseAccountOptions;
    private List<AccountOptions> incomeAccountOptions;

    public Currency getCurrency() {
        return currency;
    }

    public void setCurrency(Currency currency) {
        this.currency = currency;
    }

    public InterestType getInterestCompoundingPeriodType() {
        return interestCompoundingPeriodType;
    }

    public void setInterestCompoundingPeriodType(InterestType interestCompoundingPeriodType) {
        this.interestCompoundingPeriodType = interestCompoundingPeriodType;
    }

    public InterestType getInterestPostingPeriodType() {
        return interestPostingPeriodType;
    }

    public void setInterestPostingPeriodType(InterestType interestPostingPeriodType) {
        this.interestPostingPeriodType = interestPostingPeriodType;
    }

    public InterestType getInterestCalculationType() {
        return interestCalculationType;
    }

    public void setInterestCalculationType(InterestType interestCalculationType) {
        this.interestCalculationType = interestCalculationType;
    }

    public InterestType getInterestCalculationDaysInYearType() {
        return interestCalculationDaysInYearType;
    }

    public void setInterestCalculationDaysInYearType(InterestType interestCalculationDaysInYearType) {
        this.interestCalculationDaysInYearType = interestCalculationDaysInYearType;
    }

    public InterestType getAccountingRule() {
        return accountingRule;
    }

    public void setAccountingRule(InterestType accountingRule) {
        this.accountingRule = accountingRule;
    }

    public List<Currency> getCurrencyOptions() {
        return currencyOptions;
    }

    public void setCurrencyOptions(List<Currency> currencyOptions) {
        this.currencyOptions = currencyOptions;
    }

    public List<InterestType> getInterestCompoundingPeriodTypeOptions() {
        return interestCompoundingPeriodTypeOptions;
    }

    public void setInterestCompoundingPeriodTypeOptions(List<InterestType> interestCompoundingPeriodTypeOptions) {
        this.interestCompoundingPeriodTypeOptions = interestCompoundingPeriodTypeOptions;
    }

    public List<InterestType> getInterestPostingPeriodTypeOptions() {
        return interestPostingPeriodTypeOptions;
    }

    public void setInterestPostingPeriodTypeOptions(List<InterestType> interestPostingPeriodTypeOptions) {
        this.interestPostingPeriodTypeOptions = interestPostingPeriodTypeOptions;
    }

    public List<InterestType> getInterestCalculationTypeOptions() {
        return interestCalculationTypeOptions;
    }

    public void setInterestCalculationTypeOptions(List<InterestType> interestCalculationTypeOptions) {
        this.interestCalculationTypeOptions = interestCalculationTypeOptions;
    }

    public List<InterestType> getInterestCalculationDaysInYearTypeOptions() {
        return interestCalculationDaysInYearTypeOptions;
    }

    public void setInterestCalculationDaysInYearTypeOptions(List<InterestType> interestCalculationDaysInYearTypeOptions) {
        this.interestCalculationDaysInYearTypeOptions = interestCalculationDaysInYearTypeOptions;
    }

    public List<InterestType> getLockinPeriodFrequencyTypeOptions() {
        return lockinPeriodFrequencyTypeOptions;
    }

    public void setLockinPeriodFrequencyTypeOptions(List<InterestType> lockinPeriodFrequencyTypeOptions) {
        this.lockinPeriodFrequencyTypeOptions = lockinPeriodFrequencyTypeOptions;
    }

    public List<InterestType> getAccountingRuleOptions() {
        return accountingRuleOptions;
    }

    public void setAccountingRuleOptions(List<InterestType> accountingRuleOptions) {
        this.accountingRuleOptions = accountingRuleOptions;
    }

    public List<PaymentTypeOption> getPaymentTypeOptions() {
        return paymentTypeOptions;
    }

    public void setPaymentTypeOptions(List<PaymentTypeOption> paymentTypeOptions) {
        this.paymentTypeOptions = paymentTypeOptions;
    }

    public List<InterestType> getWithdrawalFeeTypeOptions() {
        return withdrawalFeeTypeOptions;
    }

    public void setWithdrawalFeeTypeOptions(List<InterestType> withdrawalFeeTypeOptions) {
        this.withdrawalFeeTypeOptions = withdrawalFeeTypeOptions;
    }

    public AccountOptions getLiabilityAccountOptions() {
        return liabilityAccountOptions;
    }

    public void setLiabilityAccountOptions(AccountOptions liabilityAccountOptions) {
        this.liabilityAccountOptions = liabilityAccountOptions;
    }

    public List<AccountOptions> getAssetAccountOptions() {
        return assetAccountOptions;
    }

    public void setAssetAccountOptions(List<AccountOptions> assetAccountOptions) {
        this.assetAccountOptions = assetAccountOptions;
    }

    public List<AccountOptions> getExpenseAccountOptions() {
        return expenseAccountOptions;
    }

    public void setExpenseAccountOptions(List<AccountOptions> expenseAccountOptions) {
        this.expenseAccountOptions = expenseAccountOptions;
    }

    public List<AccountOptions> getIncomeAccountOptions() {
        return incomeAccountOptions;
    }

    public void setIncomeAccountOptions(List<AccountOptions> incomeAccountOptions) {
        this.incomeAccountOptions = incomeAccountOptions;
    }

    @Override
    public String toString() {
        return "SavingProductsTemplate{" +
                "currency=" + currency +
                ", interestCompoundingPeriodType=" + interestCompoundingPeriodType +
                ", interestPostingPeriodType=" + interestPostingPeriodType +
                ", interestCalculationType=" + interestCalculationType +
                ", interestCalculationDaysInYearType=" + interestCalculationDaysInYearType +
                ", accountingRule=" + accountingRule +
                ", currencyOptions=" + currencyOptions +
                ", interestCompoundingPeriodTypeOptions=" + interestCompoundingPeriodTypeOptions +
                ", interestPostingPeriodTypeOptions=" + interestPostingPeriodTypeOptions +
                ", interestCalculationTypeOptions=" + interestCalculationTypeOptions +
                ", interestCalculationDaysInYearTypeOptions=" + interestCalculationDaysInYearTypeOptions +
                ", lockinPeriodFrequencyTypeOptions=" + lockinPeriodFrequencyTypeOptions +
                ", withdrawalFeeTypeOptions=" + withdrawalFeeTypeOptions +
                ", paymentTypeOptions=" + paymentTypeOptions +
                ", accountingRuleOptions=" + accountingRuleOptions +
                ", liabilityAccountOptions=" + liabilityAccountOptions +
                ", assetAccountOptions=" + assetAccountOptions +
                ", expenseAccountOptions=" + expenseAccountOptions +
                ", incomeAccountOptions=" + incomeAccountOptions +
                '}';
    }
}

File: core/database/src/main/java/com/mifos/room/entities/templates/savings/SavingProductsTemplate.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.room.entities.templates.savings

import android.os.Parcelable
import com.mifos.core.entity.accounts.savings.Currency
import com.mifos.core.model.objects.account.saving.FieldOfficerOptions
import com.mifos.core.model.objects.commonfiles.InterestType
import com.mifos.core.model.objects.template.saving.AccountOptions
import com.mifos.room.entities.PaymentTypeOption
import kotlinx.parcelize.IgnoredOnParcel
import kotlinx.parcelize.Parcelize

/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
/**
 * Created by rajan on 13/3/16.
 */
@Parcelize
class SavingProductsTemplate(
    var currency: Currency? = null,

    @IgnoredOnParcel
    var interestCompoundingPeriodType: InterestType? = null,

    @IgnoredOnParcel
    var interestPostingPeriodType: InterestType? = null,

    @IgnoredOnParcel
    var interestCalculationType: InterestType? = null,

    @IgnoredOnParcel
    var interestCalculationDaysInYearType: InterestType? = null,

    @IgnoredOnParcel
    var accountingRule: InterestType? = null,

    var currencyOptions: List<Currency>? = null,

    @IgnoredOnParcel
    var interestCompoundingPeriodTypeOptions: List<InterestType>? = null,

    @IgnoredOnParcel
    var interestPostingPeriodTypeOptions: List<InterestType>? = null,

    @IgnoredOnParcel
    var interestCalculationTypeOptions: List<InterestType>? = null,

    @IgnoredOnParcel
    var interestCalculationDaysInYearTypeOptions: List<InterestType>? = null,

    @IgnoredOnParcel
    var lockinPeriodFrequencyTypeOptions: List<InterestType>? = null,

    @IgnoredOnParcel
    var withdrawalFeeTypeOptions: List<InterestType>? = null,

    var paymentTypeOptions: List<PaymentTypeOption>? = null,

    @IgnoredOnParcel
    var accountingRuleOptions: List<InterestType>? = null,

    @IgnoredOnParcel
    var liabilityAccountOptions: AccountOptions? = null,

    @IgnoredOnParcel
    var assetAccountOptions: List<AccountOptions>? = null,

    @IgnoredOnParcel
    var expenseAccountOptions: List<AccountOptions>? = null,

    @IgnoredOnParcel
    var incomeAccountOptions: List<AccountOptions>? = null,

    @IgnoredOnParcel
    var fieldOfficerOptions: List<FieldOfficerOptions>? = null,
) : Parcelable
